met-web/src/components/engagement/old-view/widgets/Video/VideoWidgetView.tsx

```diff
@@ -69,35 +142,126 @@ const VideoWidgetView = ({ widget }: VideoWidgetProps) => {
+            {/* Overlay covers play button for Mixcloud so it shouldn't be displayed */}
+            <When condition={showOverlay && 'Mixcloud' !== getVideoSource(videoWidget.video_url)}>
+                <Grid item xs={12} sx={{ maxHeight: '0px', padding: '0 !important', margin: '0' }}>
+                    <VideoOverlay
+                        videoOverlayTitle={videoOverlayTitle}
+                        source={getVideoSource(videoWidget.video_url)}
+                    />
                 </Grid>
+            </When>
+            <Grid item xs={12} sx={{ paddingTop: '0 !important', marginTop: '0' }}>
+                <AspectRatioContainer style={{ borderRadius: '16px', overflow: 'hidden' }}>
+                    <ReactPlayer
+                        onDuration={(duration) => {
+                            setVideoDuration(formatVideoDuration(duration));
+                        }}
+                        url={videoWidget.video_url}
+                        controls
+                        onReady={(player) => {
+                            const videoSource = getVideoSource(videoWidget.video_url);
+                            setVideoOverlayTitle(getVideoTitle(player, videoSource, videoWidget.title));
+                        }}
+                        onPlay={() => setShowOverlay(false)}
+                        width="100%"
+                        height={'100%'}
+                        config={playerConfig}
+                        style={{
+                            position: 'absolute',
+                            top: 0,
+                            left: 0,
+                            width: '100%',
+                            height: '100%',
+                        }}
+                    />
+                </AspectRatioContainer>
+            </Grid>
+            <Grid item xs={12} sx={{ pt: '0.5rem !important' }}>
+                <BodyText
+                    style={{ fontSize: '0.95rem', color: isDarkMode ? colors.surface.white : Palette.text.primary }}
+                >
+                    {videoDuration}
+                </BodyText>
             </Grid>
-        </MetPaper>
+        </Grid>
+    );
+};
+
+interface VideoOverlayProps {
+    videoOverlayTitle: string;
+    source: string;
+}
+
+const VideoOverlay = ({ videoOverlayTitle, source }: VideoOverlayProps) => {
+    const getIcon = (source: string) => {
+        if ('YouTube' === source) {
+            return faYoutube;
+        } else if ('Vimeo' === source) {
+            return faVimeo;
+        } else if ('Facebook' === source) {
+            return faFacebookF;
+        } else if ('Twitch' === source) {
+            return faTwitch;
+        } else if ('SoundCloud' === source) {
+            return faSoundcloud;
+        } else if ('Mixcloud' === source) {
+            return faMixcloud;
+        } else if ('DailyMotion' === source) {
+            return faDailymotion;
+        } else {
+            return faQuestionCircle;
+        }
+    };
```

Comment:
Again, please replace this large conditional with an object map or other solution. I recommend using the same map to get the source name and icon because they are related 1-1

Reply:
Used the same object array from before to solve this issue.